### ISSUE TITLE
Package configuration errors are not shown when emacs starts

### ISSUE THREAD

**Issue by UndeadKernel** [hlissner/doom-emacs#319]
When starting emacs with an error inside a (def-package!) config or init, the error will be deleted from the message log and will not be shown. Furthermore, the package itself will not be loaded. The user will not be told that there was an error or that the package failed to load.

To reproduce, insert
```
(error "Panik attack!")
```
in, for example, modules/ui/doom/config.el at line 6, so that the file looks like this:
```
(def-package! doom-themes
  :config
  (error "Panik attack!")
  (unless doom-theme
    (setq doom-theme 'doom-one)
   ...
```

Start emacs as normally. Notice how the error message is not shown and how the UI of doom is not properly displayed. The user is left with no easy way to detect and correct the error.

**Comment by hlissner** [comment]
Could you check your buffer list and see if `*Warnings*` exists? It could be that the messages are being reported _but_ this window isn't properly being displayed.

**Comment by UndeadKernel** [comment]
The errors do appear in the *Warnings* buffer. The window, as you assumed, is not displayed.
I also noticed that, due to only showing files of the same "perspective", the *Warnings* buffer is not easily accessible. I needed to run (list-buffers) to see that buffer.

**Comment by hlissner** [comment]
I can't reproduce this on master or develop. I've pushed something that _could_ address this issue in develop however. Let's try a couple things:

+ Could you disable `:feature workspaces` to see if this fixes the warnings buffer?
+ Could you try the latest `develop` and see if that has fixed your problem?

**Comment by UndeadKernel** [comment]
I've tested the bug using a fresh version of DOOM emacs from the develop branch; I only copied init.el.example to init.el.

The bug is present, always reproducible: The warning buffer is not displayed when emacs starts (if there were errors in a module).

The latest version in the develop branch is still affected.

If I remove the "workspaces" module, the \*Warnings\* buffer **is** shown correctly (both, in the master and develop branch).
